cli/command/context/list_test.go

```diff
@@ -0,0 +1,59 @@
+package context
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/formatter"
+	"gotest.tools/assert"
+	"gotest.tools/env"
+	"gotest.tools/golden"
+)
+
+func createTestContextWithKubeAndSwarm(t *testing.T, cli command.Cli, name string) {
+	revert := env.Patch(t, "KUBECONFIG", "./testdata/test-kubeconfig")
+	defer revert()
+
+	err := runCreate(cli, &createOptions{
+		name:                     name,
+		defaultStackOrchestrator: "all",
+		description:              "description of " + name,
+		kubernetes: kubernetesEndpointOptions{
+			fromCurrent: true,
+		},
+		docker: dockerEndpointOptions{
+			host: "https://someswarmserver",
+		},
+	})
+	assert.NilError(t, err)
+}
+
+func TestList(t *testing.T) {
+	buf := bytes.NewBuffer(nil)
+	cli, cleanup, err := makeCliMock(t, withOut(buf))
+	assert.NilError(t, err)
+	defer cleanup()
+	createTestContextWithKubeAndSwarm(t, cli, "current")
+	createTestContextWithKubeAndSwarm(t, cli, "other")
+	cli.currentContext = "current"
+	assert.NilError(t, runList(cli, &listOptions{
+		format: formatter.TableFormatKey,
+	}))
+	golden.Assert(t, buf.String(), "list.golden")
+}
+
+func TestListQuiet(t *testing.T) {
+	buf := bytes.NewBuffer(nil)
+	cli, cleanup, err := makeCliMock(t, withOut(buf))
+	assert.NilError(t, err)
+	defer cleanup()
+	createTestContextWithKubeAndSwarm(t, cli, "current")
+	createTestContextWithKubeAndSwarm(t, cli, "other")
+	cli.currentContext = "current"
+	assert.NilError(t, runList(cli, &listOptions{
+		format: formatter.TableFormatKey,
+		quiet:  true,
+	}))
+	assert.Equal(t, "current\nother\n", buf.String())
```

Comment:
You can also use a golden here 😄

Reply:
really, for 2 lines ?

Reply:
https://github.com/docker/cli/blob/master/cli/command/service/testdata/service-list-sort.golden
Ok there are 3 lines, but still... 😄

Reply:
@simonferquel we have some golden file with 2 lines :angel: :trollface: